tensorboard/webapp/metrics/store/metrics_reducers.ts

```diff
@@ -1050,6 +1052,17 @@ const reducer = createReducer(
     let nextStepSelectorEnabled = state.stepSelectorEnabled;
     let linkedTimeSelection = state.linkedTimeSelection;
 
+    const nextCardStateMap = Object.entries(state.cardStateMap).reduce(
+      (cardStateMap, [cardId, cardState]) => {
+        cardStateMap[cardId] = {
+          ...cardState,
+          rangeSelection: CardSelectionState.GLOBAL,
```

Comment:
I think a comment here is worthwhile. It took even me some time to wrap my head around the logic and I'm the one that introduced this complication.

Could say something like:
Since range selection was last toggled on/off, some cards were individually turned off/on respectively.  Those cards differed from the "global" step selection enablement state. Now that range selection is being turned back on or off, all cards once again have the "global" state.